Onboarding 누락된 기능 추가

agvber opened the pull request:
## 🤷‍♂️ Description
- [x] 카카오 로그인이 정상작동하는지 확인합니다
- [x] 온보딩 화면 순서가 맞는지 체크합니다.
- [x] 그룹 생성이 정상적으로 작동하는지 체크합니다.
- [x] 그룹 가입이 정상적으로 작동하는지 체크합니다.
- [x] 프로필 수정이 정상적으로 작동하는지 체크합니다.
- [x] 알림 권한 체크 화면에서 정상 장독하는지 체크합니다.
- [x] 온보딩 중간에 앱 종료 후 앱 재시작시 온보딩 화면을 다시 불러오도록 합니다.
- [ ] 로그인 완료후 화면이동이 부자연스러운 부분을 수정합니다.

## 🔄 How to reproduce bug
- 어떻게 해결하였는지 간략하게 설명합니다.

## 📸 Screenshots
- 필요한 경우 사진을 남깁니다.

## ✳️Remarks
- addOnboardingMakeInviteGrandParentsScreen 해당화면은 기획이 명확하지 않아 주석처리 하였습니다 추후에 수정하도록 하겠습니다.
- 프로필 업로드가 ViewModel onCleared가 될때 API 호출되는데 그부분이 사실 무서운데 그부분 제외하면 문제 없습니다

agvber committed:
fix: 카카오 로그인 버튼이 보이지 않는 문제 수정

agvber committed:
feat: Onboarding 그룹 생성 API 요청 기능 추가

agvber committed:
feat: 온보딩 과정에서 사용자가 그룹 가입이 필요한 경우의 분기 처리

agvber committed:
feat: Onboarding 초대 코드 입력 화면에서 서버 API 요청 기능 추가

agvber committed:
fix: 그룹 생성 화면에서 예외 처리가 정상적이지 않는 문제 수정

Event를 StateFlow를 사용하여 그룹 생성 후에 Success 상태를 유지하므로 성격상 맞지 않으므로 수정하였습니다.

agvber committed:
fix: 온보딩 알림 체크화면에서 하단 버튼이 보이지 않는 문제를 수정하였습니다. (구형기기 한정)

agvber committed:
feat: 온보딩 권한 체크 화면에서 사용자가 권한 허용하지 않는 경우 경고 메시지를 추가하였습니다.

agvber committed:
fix: 오보딩 권한 체크 후 다음 화면으로 넘어가지 않는 문제 수정하였습니다.

agvber committed:
feat: [onboarding] 사용자가 이미 그룹이 있는 경우 온보딩 과정을 skip 할수 있도록 변경하였습니다.

agvber committed:
feat: [onboarding] 그룹 가입이 되어있지 않는 사용자는 온보딩 화면으로 이동 할 수 있도록 변경

agvber committed:
feat: [onboarding] 초기 로드 데이터가 정상적이지 않는 경우 로그인 화면으로 이동

agvber committed:
feat: [onboarding] Profile 수정 API 화면 연결

agvber committed:
refactor: modify access modifier from public to internal

agvber committed:
refactor: modify code design

theBettor reviewed (approved):
고생하셨습니다 좋은 공부가 될거같아요

agvber committed:
fix: 앱 시작 시 사용자가 온보딩을 완료하였는지 체크하는 과정에서 에러를 수정하였습니다.

온보딩을 완료하지 못하고 이탈하면 다음번에 앱 실행 시 그룹 생성으로 바로 이동할 수 있도록 수정했지만, 로그인이 필요한 사용자도 체크하는 문제가 있었습니다.
